### DISCUSSION TITLE
internal Message<T> DTO

### DISCUSSION THREAD

**Discussion by tomliversidge** [fsharplang-ru/pulsar-client-dotnet#269] [General]
Hi,

I've just started working with the library - thanks for this!

One question I had was regarding the internal nature of `Message<T>` - I have code that is consuming a message, which I then want to pass into my own classes and methods, but the fact that it is internal is preventing me being able to create these messages in unit tests (i'm using C# if that makes any difference)

Does anyone face a similar issue?

```
public class MyClass {

    public bool MyMethod<T>(Message<T> message)
    {
        // do something with message
    }
}
```
Is there any way to create a new `Message<T>` outside of the library itself?

**Comment by Lanayx** [comment]
Hi, @tomliversidge ! The `Message` is actually not internal, only it's constructor is, this was done on purpose - to prevent users misusing it. For unit tests I belive you can create Message using reflection to bypass internal restriction, however I'd recommend you create your own DTO with fields mapped from `Message` (I suspect you won't need them all) for passing it between your classes and business logic. This will guarantee that in case of any changes in library `Message` you'll only have to update one mapping and nothing else.
